tests/difference/core/quint_model/ccv.qnt

```diff
@@ -0,0 +1,877 @@
+// -*- mode: Bluespec; -*-
+module CCVTypes {
+    import Time.* from "./libraries/Time"
+
+    type Node = str
+    type Chain = str
+    type Power = int
+    type VSCId = int
+    type ValidatorSet = Node -> Power
+    type Height = int
+    // a list of validator sets per blocks, ordered by recency
+    type VotingPowerHistory = List[ValidatorSet]
+
+    type VSCPacket =
+        {   
+            // the identifier for this packet
+            id: VSCId,
+            // the new validator set. in the implementation, this would be a list of validator updates
+            validatorSet: ValidatorSet,
+            // the time at which the packet was sent. used to check whether packets have timed out.
+            sendingTime: Time
+        }
+
+    type VSCMaturedPacket = 
+        {
+            // the id of the VSCPacket that matured
+            id: VSCId,
+            // the time at which the packet was sent. used to check whether packets have timed out.
+            sendingTime: Time
+        }
+
+
+    // state that each chain needs to store, whether consumer or provider.
+    type ChainState = {
+        // Stores the list of voting powers that corresponded to voting powers 
+        // at blocks over the chains entire existence.
+        // Voting powers should be ordered by recency in descending order.
+        votingPowerHistory: VotingPowerHistory,
+
+        // the current validator set on each chain.
+        // this will be included in the next block, but might not be final yet,
+        // e.g. there may be more modifications in the current block.
+        currentValidatorSet: ValidatorSet,
+
+        // the latest timestamp that was comitted on chain
+        lastTimestamp: Time,
+    }
+
+    // utility function: returns a chain state that is initialized minimally.
+    pure def GetEmptyChainState(): ChainState = 
+        {
+            votingPowerHistory: List(),
+            currentValidatorSet: Map(),
+            lastTimestamp: 0,
+        }
+
+    // Defines the current state of the provider chain. Essentially, all information here is stored by the provider on-chain (or could be derived purely by information that is on-chain).
+    type ProviderState =
+        {
+            // the state that each chain needs to store
+            chainState: ChainState,
+
+            // Stores, for each consumer chain, the list of packets that have been sent to the consumer chain
+            // and have not been received yet.
+            // In the implementation, this would roughly be the unacknowledged packets on an ibc channel.
+            outstandingPacketsToConsumer: Chain -> List[VSCPacket],
```

Comment:
I tried to make it clearer. Outstanding are not yet received by the consumer, sent are ones that we have not yet received a maturation for.
If you have a suggestion, this could be simplified, but at least my intuition is that any solution should use two separate lists to do this

Reply:
So, `outstandingPacketsToConsumer` is just a way to put a packet in a send queue that would be eventually be picked up and delivered to consumer's `receivedVSCPackets`?

The idea of keeping the `sentVSCPackets` is to check if we haven't received a `VSCMaturedPacket` since sending the `VSCPacket` and this cannot be done with `outstandingPacketsToConsumer` because you pop packets from `outstandingPacketsToConsumer` after sending them?

It might make sense to rename `sentVSCPackets` to `sentVSCPacketsToConsumer` similarly to `outstandingPacketsToConsumer` in such a case?

Reply:
correct. Changed according to the suggestion, thanks